sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/implementation/HttpUrlConnectionResponse.java

```diff
@@ -0,0 +1,78 @@
+package com.azure.core.http.httpurlconnection.implementation;
+
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+
+public class HttpUrlConnectionResponse extends HttpResponse {
```

Comment:
This type seems to only be used in the main package. Perhaps it can be moved there, and made package-private? That reduces the API visibility somewhat.

Reply:
HttpUrlConnectionResponse is also present in the SocketClient class, meaning this can't be actioned.